### PR TITLE
Audio segments

### PR DESCRIPTION
**Pull request by sharabash**
@mmahalwy needs security

### CODE REVIEW

**Comment by mmahalwy** [comment]
I'd prefer if this was part of the v2 api. seems like the perfect candidate for it no?

**Comment by mmahalwy** [comment]
also, we should model the segment

**Comment by sharabash** [comment]
i tried modeling the segment but it wasnt worth it at the end of the day,
it just made things overly complex with no added value; it makes more sense
to store it as a single json text column in the file table

**Comment by sharabash** [comment]
and yeah i think this would be a good candidate for the v2 api

**Comment by mmahalwy** [comment]
yeah v2 is heavy on the modeling. if you stored it as json, then it's fine, no model needed :)

File: app/models/audio/file.rb [hunk 1]
```diff
@@ -22,13 +22,31 @@ class Audio::File < ActiveRecord::Base
     has_one :reciter, class_name: 'Audio::Reciter', through: :recitation
 
     def self.bucket_audio(audio_id, keys)
+
```
Comment (mmahalwy) [inline app/models/audio/file.rb]: this method is not being used anymore as we now use V2 on the frontend.

File: app/models/audio/file.rb [hunk 2]
```diff
@@ -22,13 +22,31 @@ class Audio::File < ActiveRecord::Base
...
+        require 'base64'
+        require 'gibberish'
+        require 'json'
```
Comment (mmahalwy) [inline app/models/audio/file.rb]: requires in methods = bad

**Comment by mmahalwy** [comment]
i dont think we should complicate it with a new github repo, personally. especially since it's just a string that we can store somewhere and tell someone to do it themselves no?

For example, it could be an `.env` variable that we can give to people we trust and since it's an env variable, you can then have it passed in with docker-compose in quran-apps

**Comment by sharabash** [comment]
that makes sense, what env var name do you suggest and do we store it in
development.env or what?

**Comment by sharabash** [comment]
@mmahalwy i updated this to the state that its as ready as it can get without updating it to use the v2 controller. can you do that?

**Comment by mmahalwy** [comment]
closing for #42

**Pull request closed without merging**